compiler/rustc_codegen_ssa/src/back/linker.rs

```diff
@@ -511,9 +511,10 @@ impl<'a> Linker for GccLinker<'a> {
             self.cmd.arg(path);
         } else if self.sess.target.is_like_osx {
             self.linker_arg("-force_load");
-            self.linker_arg(&path);
+            self.linker_arg(path);
         } else {
-            self.linker_args(&[OsString::from("--whole-archive"), path.into()]);
+            self.linker_arg("--whole-archive");
+            self.cmd.arg(path);
```

Comment:
why is this not a linker arg?

Reply:
For consistency with `!whole_archive` case above, whether `-Wl,` is added doesn't matter in this context.

We could probably consistently add it in all cases though, to save C compiler some work? (Looking into files and determining that they should indeed be passed through to the linker.)

Reply:
>We could probably consistently add it in all cases though, to save C compiler some work

This would be some larger work, so I've just reverted this change for now.

Reply:
>to save C compiler some work? (Looking into files and determining that they should indeed be passed through to the linker.)

Update: gcc doesn't even inspect the passed .o and .a files, it just looks at file extensions.
clang does call fstat on them, but doesn't inspect the contents either.